internal/pkg/daemon/enricher/container.go

```diff
@@ -124,6 +120,10 @@ func (e *Enricher) getContainerInfo(
 						recordProfile: recordProfile,
 					}
 
+					if e.labelProblematicContainers {
+						go labelContainer(e.impl, e.clientset, *info, e.logger)
```

Comment:
I think this can be racy. If a pod is already labeled or about to be labeled it should be skipped.